core/server/master/src/main/java/alluxio/master/file/replication/ReplicationChecker.java

```diff
@@ -137,15 +136,27 @@ public void close() {
     // Nothing to clean up
   }
 
+  private boolean checkCorrectlyReplicated(InodeFile file, BlockInfo blockInfo) {
+    Set<String> pinnedMediumTypes = file.getMediumTypes();
+    if (blockInfo == null && !pinnedMediumTypes.isEmpty()) {
+      return false;
+    }
+    Set<String> storedMediumTypes = new HashSet<>();
+    for (BlockLocation loc : blockInfo.getLocations()) {
+      storedMediumTypes.add(loc.getMediumType());
+    }
+    // if all pinned medium type is found in block locations, then it is correctly pinned.
+    return storedMediumTypes.containsAll(pinnedMediumTypes);
```

Comment:
Couldn't we simplify the logic to just fail fast if we find a medium type which is not contained?

Reply:
we still need to build the set though right? i am not sure if the logic is simpler?

Reply:
Oh hmm shouldn't this method return more than a boolean then? Because we need to know which blocks are not correctly replicated?